aptos-move/framework/aptos-framework/sources/jwks.move

```diff
@@ -0,0 +1,480 @@
+        observation_interval_ms: u64,
+    }
+
+    /// An observed but not yet supported JWK.
+    struct UnsupportedJWK has copy, drop, store {
+        payload: vector<u8>,
+    }
+
+    /// A JWK where `kty` is `RSA`.
+    struct RSA_JWK has copy, drop, store {
+        kid: String,
+        kty: String,
+        alg: String,
+        e: String,
+        n: String,
+    }
+
+    /// A JWK.
+    ///
+    /// Currently supported `content` types:
+    /// - `RSA_JWK`
+    /// - `UnsupportedJWK`
+    struct JWK has copy, drop, store {
+        id: vector<u8>,
+        content: Any,
+    }
+
+    /// A provider and its JWKs.
+    struct ProviderJWKSet has copy, drop, store {
+        /// The utf-8 encoding of the issuer string (e.g., "https://accounts.google.com").
+        issuer: vector<u8>,
+
+        /// The `jwks` should each have a unique `id`, and should be sorted by `id` in alphabetical order.
+        jwks: vector<JWK>,
+    }
+
+    /// All OIDC providers and their JWK sets.
+    struct JWKMap has copy, drop, store {
+        /// Entries should each have a unique `issuer`, and should be sorted by `issuer` in an alphabetical order.
+        entries: vector<ProviderJWKSet>,
+    }
+
+    /// A `JWKMap` maintained by JWK consensus.
+    struct OnChainJWKMap has copy, drop, key, store {
+        version: u64,
+        jwk_map: JWKMap,
+    }
+
+    #[event]
+    /// When an on-chain JWK set update is done, this event is sent to reset the JWK consensus state in all validators.
+    struct OnChainJWKMapUpdated has drop, store {
+        epoch: u64,
+        version: u64,
+        jwk_map: JWKMap,
+    }
+
+    /// A small edit that can be applied to a `JWKMap`.
+    ///
+    /// Currently supported `content` types:
+    /// - `JWKMapEditCmdDelAll`
+    /// - `JWKMapEditCmdDelIssuer`
+    /// - `JWKMapEditCmdDelJwk`
+    /// - `JWKMapEditCmdPutJwk`
+    struct JWKMapEdit has drop, store {
+        content: Any,
+    }
+
+    struct JWKMapEditCmdDelAll has copy, drop, store {}
+
+    struct JWKMapEditCmdDelIssuer has copy, drop, store {
+        issuer: vector<u8>,
+    }
+
+    struct JWKMapEditCmdDelJwk has copy, drop, store {
+        issuer: vector<u8>,
+        jwk_id: vector<u8>,
+    }
+
+    struct JWKMapEditCmdPutJwk has copy, drop, store {
+        issuer: vector<u8>,
+        jwk: JWK,
+    }
+
+    /// A sequence of `JWKMapEdit` that needs to be applied *one by one* to the JWK consensus-maintained `JWKMap` before being used.
+    ///
+    /// Maintained by governance proposals.
+    struct JWKMapPatch has key {
+        edits: vector<JWKMapEdit>,
+    }
+
+    //
+    // Structs end.
+    // Functions begin.
+    //
+
+    /// Initialize some JWK resources. Should only be invoked by genesis.
+    public(friend) fun initialize(account: &signer) {
+        system_addresses::assert_aptos_framework(account);
+        move_to(account, OIDCProviderSet { providers: vector[] });
+        move_to(account, jwk_consensus_config_v0(10000));
```

Comment:
Make `10_000` into a `DEFAULT_OBSERVATION_INTERVAL_MS` constant.

Reply:
This entry is gone - fetch interval should be a node config, instead of an on-chain one.